src/components/DatePickerRange.react.js

```diff
@@ -0,0 +1,306 @@
+        focusedInput: props.autoFocusEndDate ? momentProps.startDate : momentProps.endDate,
+        startDate: momentProps.startDate,
+        endDate: momentProps.endDate,
+        initialVisibleMonth: momentProps.initialVisibleMonth,
+        minDateRange: momentProps.min,
+        maxDateRange: momentProps.max,
+        initialStartDate: this.props.start_date,
+        initialEndDate: this.props.end_date,
+        firstInitialVisibleMonth: this.props.initial_visible_month,
+        initialMinDateRange: this.props.min_date_range,
+        initialMaxDateRange: this.props.max_date_range
+      };
+    }
+
+    convertPropsToMoment(props) {
+      let startDate, endDate = null;
+      let initialVisibleMonth = moment(props.startDate);
+      if (typeof props.startDate !== 'undefined') {
+        startDate = moment(props.startDate);
+      }
+      if (typeof props.endDate !== 'undefined') {
+        endDate = moment(props.endDate);
+      }
+      if (typeof props.initialVisibleMonth !== 'undefined') {
+        initialVisibleMonth = moment(props.initialVisibleMonth);
+      }
+      let min, max;
+      if (typeof props.minDateRange !== 'undefined' &&
+          typeof props.maxDateRange !== 'undefined') {
+        min = moment(props.minDateRange);
+        max = moment(props.maxDateRange);
+      }
+      try {
+        if (startDate.isAfter(endDate)) {
+          endDate = null;
+        }
+      } catch (TypeError) {
+        // Continue regardless of error
+      }
+
+      return {startDate, endDate, initialVisibleMonth, min, max}
+    }
+
+    componentWillReceiveProps(newProps) {
+      let propObj = {
+        startDate: newProps.start_date,
+        endDate: newProps.end_date,
+        initialVisibleMonth: newProps.initial_visible_month,
+        minDateRange: newProps.min_date_range,
+        maxDateRange: newProps.max_date_range
+      }
+      const momentProps = this.convertPropsToMoment(propObj);
+      if (this.state.initialStartDate != newProps.start_date) {
+        this.setState({
+          initialStartDate: newProps.start_date,
+          startDate: momentProps.startDate
+        });
+      }
+      if (this.state.initialEndDate != newProps.end_date) {
+        this.setState({
+          initialEndDate: newProps.end_date,
+          endDate: momentProps.endDate
+        });
+      }
+      if (this.state.firstInitialVisibleMonth != newProps.initial_visible_month){
+        this.setState({
+          firstInitialVisibleMonth: newProps.initial_visible_month,
+          initialVisibleMonth: momentProps.initialVisibleMonth
+        })
+      }
+      if (this.state.initialMinDateRange != newProps.min_date_range){
+        this.setState({
+          initialMinDateRange: newProps.min_date_range,
+          minDateRange: momentProps.minDateRange
+        })
+      }
+      if (this.state.initialMaxDateRange != newProps.max_date_range){
+        this.setState({
+          initialMaxDateRange: newProps.max_date_range,
```

Comment:
The values `initialMaxDateRange` and the rest are being used in the if statements to determine whether the new callback value has changed them. So when a callback changes the value of max_date_range we check to see whether it has changed with regard to its previous value `initialMaxDateRange` and if so we update the state.

I agree it is a pretty confusing way to set it up though, I've changed the variable names to `prevMaxDateAllowed` to make it less confusing